### PR TITLE
Fix bug in dss_local_ghost_kernel!

### PR DESCRIPTION
**Pull request by sriharshakandala**
Fix bug in dss_local_ghost_kernel!

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by sriharshakandala** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try-

**Comment by sriharshakandala** [comment]
bors try

File: src/Spaces/dss_cuda.jl [hunk 1]
```diff
@@ -391,25 +391,25 @@ function dss_local_ghost_kernel!(
 ) where {FT <: AbstractFloat, Nq}
     gidx = threadIdx().x + (blockIdx().x - 1) * blockDim().x
     (nlevels, nperimeter, nfidx, _) = size(pperimeter_data)
-    nghostvertices = length(ghost_vertices)
+    nghostvertices = length(ghost_vertex_offset) - 1
```
[the remark this answers is not in the document]
Comment (sriharshakandala) [reply src/Spaces/dss_cuda.jl:394]: `ghost_vertex_offset` for `ghost_vertex` goes from `ghost_vertex_offset[vertno]` to `ghost_vertex_offset[vertno+1] - 1`

File: src/Spaces/dss_cuda.jl [hunk 2]
```diff
@@ -391,25 +391,25 @@ function dss_local_ghost_kernel!(
...
     if gidx ≤ nlevels * nfidx * nghostvertices
```
[the remark this answers is not in the document]
Comment (sriharshakandala) [reply src/Spaces/dss_cuda.jl:395]: Are you suggesting `if gidx < nlevels * nfidx * nghostvertices + 1`

**Review by simonbyrne — APPROVED** [review]

**Comment by simonbyrne** [comment]
@sriharshakandala looks like we're getting timeouts: maybe push the `timeout_in_minutes` to 10 mins or larger?

**Pull request merged**